apis/druid/v1alpha1/druid_types.go

```diff
@@ -180,6 +180,11 @@ type DruidSpec struct {
 	// +optional
 	PodAnnotations map[string]string `json:"podAnnotations,omitempty"`
 
+	// ReplicationControllerAnnotations annotations to be populated in StatefulSet or Deployment spec.
+	// if the same key is specified at both the DruidNodeSpec level and DruidSpec level, the DruidNodeSpec ReplicationControllerAnnotations will take precedence.
+	// +optional
+	ReplicationControllerAnnotations map[string]string `json:"replicationControllerAnnotations,omitempty"`
```

Comment:
I don't think this is a good name since it is relevant only to the `Deployment` object. I would change this to only `Annotations` since this is clear enough where is it going to.
Examples of APIs like that: `Affinity`, `Tolerations`, etc..

Reply:
+1 . Lets keep it to Annotations.

Reply:
I think I disagree - we already have `podAnnotations` and `ingressAnnotations`, so calling this one just `Annotations` feels extremely vague. 

The DruidSpec / DruidNodeSpec objects define a collection of other kubernetes Resources (ingresses, services, statefulSets, deployments, configMaps, podDisruptionBudgets) and these annotations only apply to two of these resource types. Calling it `Annotations` implies that they would be applied to all resources, which isn't the case.

I'm open to discussion though, let me know what you think.

Reply:
You're right.
In that case, I would call it `WorkloadAnnotations` since Kubernetes calls this family of objects that. 
https://kubernetes.io/docs/concepts/workloads/controllers/